File: benchmark/c/cpw/sum_pairwise_products/sum_pairwise_products.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int sum_pairwise_products ( int n ) ;
int sum_pairwise_products ( int n ) {
  long int sum = 0;
  for ( int i = 1;
  i <= n;
  i ++ ) for ( int j = i;
  j <= n;
  j ++ ) sum = sum + i * j;
  return sum;
}

File: experimental data/RQ3&RQ4/translation_result/fail/c/cpw/mistral/Restart/few-shot-True/cot-ast/candidates-1/sum_pairwise_products/sum_pairwise_products.rs

use std::i32;

fn sum_pairwise_products(n: i32) -> i32 {
    let mut sum: i32 = 0;
    for i in 1..=n {
        for j in i..=n {
            sum = sum.saturating_add(i * j);
        }
    
    sum
}
